php/class-media.php

```diff
@@ -365,6 +365,34 @@ public function is_uploadable_media( $attachment_id ) {
 		return apply_filters( 'cloudinary_is_uploadable_media', $is_uploadable, $media_host );
 	}
 
+	/**
+	 * Is the media item size allowed to be uploaded.
+	 * Checks against the account limits.
+	 *
+	 * @param int $attachment_id The attachment ID.
+	 *
+	 * @return bool
+	 */
+	public function is_oversize_media( $attachment_id ) {
+		$is_oversize = false;
+		$callback    = 'get_attached_file';
+
+		if (
+			function_exists( 'wp_get_original_image_path' )
+			&& wp_attachment_is_image( $attachment_id )
+		) {
+			$callback = 'wp_get_original_image_path';
+		}
+
+		$file     = $callback( $attachment_id );
+		$max_size = ( wp_attachment_is_image( $attachment_id ) ? 'image_max_size_bytes' : 'video_max_size_bytes' );
+		if ( file_exists( $file ) && filesize( $file ) > $this->plugin->components['connect']->usage['media_limits'][ $max_size ] ) {
+			$is_oversize = true;
+		}
+
+		return $is_oversize;
+	}
```

Comment:
@pereirinha - since this is hitting the filesystem, and is called many times in a sync check cycle, perhaps add in a static variable cache against the `attachment_id`